lib/sqlalchemy/sql/elements.py

```diff
@@ -2201,7 +2241,9 @@ class TextClause(
     def _hide_froms(self) -> Iterable[FromClause]:
         return ()
 
-    def __and__(self, other):
+    def __and__(
+        self, other: _ColumnExpressionArgument[bool]
+    ) -> ColumnElement[bool]:
         # support use in select.where(), query.filter()
         return and_(self, other)
```

Comment:
This call is interpreted as returning `Any` by mypy - must be because of `and_` assignment problem later on, I commented on that [here](https://github.com/GodelTech/sqlalchemy/pull/5/files#r1065562963)